gfi.sh

```diff
@@ -14,69 +14,92 @@ html_content="<!DOCTYPE html>
     <meta charset=\"UTF-8\">
     <meta name=\"viewport\" content=\"width=device-width, initial-scale=1.0\">
     <title>GitHub Issues</title>
-    <style>
-        body {
-            font-family: Arial, sans-serif;
-        }
-        h1 {
-            color: #333;
-        }
-        ul {
-            list-style-type: none;
-            padding: 0;
-        }
-        li {
-            margin: 10px 0;
-        }
-        a {
-            text-decoration: none;
-            color: #007bff;
-        }
-        a:hover {
-            text-decoration: underline;
-        }
-    </style>
+    <link rel=\"stylesheet\" type=\"text/css\" href=\"style.css\">
 </head>
 <body>
+<a href=\"#menu\" class=\"menu-button\">Repos</a>
+<a id=\"close\"></a>
+<div class=\"content\">
 <h1>Good first issues in Ethereum core repos</h1>
 "
-#api uses gh tool now, reading should work without a key
-while IFS= read -r REPO; do
 
+# Create a menu for anchor links
+menu_content="<div id=\"menu\" class=\"menu\">
+<a href=\"#close\" class=\"close-button\">&#x2716;</a>
+<h2>Repositories</h2>"
+
+# Loop through repositories in repos.txt
+while IFS= read -r REPO; do
     if [[ $REPO == \#* ]]; then
         category=${REPO#\# }
         html_content="$html_content
     <h2>$category</h2>"
         continue
     fi
-    
 
     if [[ -z "$REPO" ]]; then
         continue
     fi
-    
+
+    repo_name=$(echo "$REPO" | awk -F '/' '{print $2}')
+    menu_content="$menu_content
+    <div class=\"menu-item\"><a href=\"#$repo_name\">$REPO</a></div>"
+
     html_content="$html_content
-    <h3>GFIs in $REPO</h3>
-    <ul id=\"issues-list-$REPO\">"
-    #api call with list of labels to pull
-    issues=$(gh issue list --repo $REPO --search 'label:"good first issue","D-good-first-issue","E-easy"' --json title --json url)
-
-    # pulls title and link from json, there is more fields to add
-    while IFS= read -r line; do
-        title=$(echo "$line" | jq -r '.title')
-        link=$(echo "$line" | jq -r '.url')
+    <h3 class="repo-header" id=\"$repo_name\">GFIs in <a href=\"https://github.com/$REPO\" target=\"_blank\">$REPO</a></h3>
+    <div id=\"issues-list-$repo_name\">"
+    
+    # API call with list of labels to pull
+    issues=$(gh issue list --repo $REPO --search 'label:"good first issue","D-good-first-issue","E-easy"' --json title,url,state,assignees 2>&1)
+
+    if [[ $(echo "$issues" | jq length) -eq 0 ]]; then
         html_content="$html_content
-        <li><a href=\"$link\" target=\"_blank\">$title</a></li>"
-    done <<< "$(echo "$issues" | jq -c '.[]')"
+        <div class=\"item\">
+            <p>No Current Good First Issues</p>
+        </div>"
+    else
+        # Process each issue and append to HTML content
+        while IFS= read -r line; do
+            title=$(echo "$line" | jq -r '.title')
+            link=$(echo "$line" | jq -r '.url')
+            state=$(echo "$line" | jq -r '.state')
```

Comment:
This is only pulling open issues anyway. I would remove the state and maybe add "Assigned / Not assigned" instead

Reply:
Ah, good point. Will do on next round.